### PR TITLE
This commit: i. adds dataservice methods to add and retrieve moments and...

### PR DESCRIPTION
**Pull request by kranrao**
... mementos, ii. adjusts dummy data to meet API endpoint expectations, and iii. sets up initial unit tests for dataservice service.

Adds moment table with pre-existing content.  Names methods to be added to service

Adjusts testing framework.  Sets up initial dataservice tests and mockdata

Saves incomplete tests for dataservice

getMementos and getMemento methods set and returning data like planned in API

Adds saveMemento functionality

dataservice methods built.  Need to persist data in file

Methods finalized.  Dummy data persists in a user session

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ gulp.task('mocha', function(done) {
 
 // new run tests
 gulp.task('karma', function() {
-  return gulp.src(paths.test)
+  return gulp.src('./foobar') //
```
Comment (omarduarte) [inline gulpfile.js:39]: Why are we using './foobar' ?

File: package.json [hunk 2]
```diff
@@ -39,7 +39,6 @@
     "karma-mocha": "^0.1.10",
     "karma-sinon": "^1.0.4",
     "mocha": "^2.1.0",
-    "ng-annotate": "^0.15.4",
```
Comment (omarduarte) [inline package.json:42]: Why did we delete ng-annotate? Did we needed a different plugin?

File: gulpfile.js (hunk 1 above)
Comment (kranrao) [inline gulpfile.js:39]: apparently there is a bug with gulp karma where the karma.config file is not read.  a work around is to set the parameter to something arbitrary. see link below:

http://stackoverflow.com/questions/22413767/angular-testing-with-karma-module-is-not-defined

File: package.json (hunk 2 above)
Comment (kranrao) [inline package.json:42]: sorry, deleted in favor of adding gulp-plugins later.  this is what Jon Pappa was using in his example.

**Pull request merged**